tests/run_tests.py

```diff
@@ -27,170 +27,99 @@ def fetch(dataset=None):
 DATA_DIR = Path('~/mne_data').expanduser()
 
 
-class TestOptionsT(TypedDict):
-    dataset: str
-    config: str
-    steps: Collection[str]
-    task: Optional[str]
-    env: Dict[str, str]
-
-
-TEST_SUITE: Dict[str, TestOptionsT] = {
-    'ds003392': {
-        'dataset': 'ds003392',
-        'config': 'config_ds003392.py',
-        'steps': ('preprocessing', 'sensor', 'report'),
-        'task': None,
-        'env': {}
-    },
-    'ds004229': {
-        'dataset': 'ds004229',
-        'config': 'config_ds004229.py',
-        'steps': ('preprocessing', 'sensor', 'report'),
-        'task': None,
-        'env': {}
+# Until PEP655 lands in 3.11 and we can use NotRequired, using this forces our
+# code to be non-DRY:
+#
+# class TestOptionsT(TypedDict):
+#     dataset: str
+#     config: str
+#     steps: Collection[str]
+#     task: Optional[str]
+#     env: Dict[str, str]
+
+
+# If not supplied below, the defaults are:
+# key: {
+#     'dataset': key.split('_')[0],
+#     'config': f'config_{key}.py',
+#     'steps': ('preprocessing', 'sensor', 'report'),
+#     'env': {},
+#     'task': None,
+# }
+#
+TEST_SUITE: Dict[str, Dict] = {
+    'ds003392': {},
+    'ds004229': {},
+    'ds004107': {},
+    'ds000117': {},
+    'ds003775': {},
+    'eeg_matchingpennies': {
+        'dataset': 'eeg_matchingpennies',
```

Comment:
I saw that a `TypedDict` was used in `datasets.py`, and this is nice syntatic sugar, but I think it hurts maintainability in the end here. Things can be simplified *a lot* and become more readable with defaults if we abandon it (at least until [PEP655 lands in 3.11 and we require it so we can use NotRequired](https://peps.python.org/pep-0655/)). By making the entries optional and having reasonable defaults for them:

1. Code is more DRY, less chance for error
2. Easier to add new datasets (see 4107 in the list above is a single easy line and everything else "just works")
3. It's immediately obvious where each dataset deviates from the default processing scheme enumerated above

So I've implemented this here.

Reply:
I disagree with these changes, as they reduce flexibility should we need it, and make it **less** clear what's going on. I specifically designed this here API to be very explicit and, yes, maybe a bit verbose, but with the advantage that no magic happens e.g. in terms of how the configuration files are located in the filesystem etc. The `dataset`, `config`, and `steps` keys should not be optional. Also, without a `TypedDict`, my IDE doesn't provide me any help as to which keys are valid. So this actually makes adding new test cases **harder**.

Interim reconsilation proposition:
Keep it a `TypedDict` but make **all** keys optional. So you can have your cake, and I can eat mine too:

```python
class TestOptionsT(TypedDict, total=False):
    ....
```

Reply:
> they reduce flexibility should we need it

Can you explain how?  I'm a bit of a TypedDict n00b so it's not obvious to me how @larsoner's proposal ties your hands here.

> and make it **less** clear what's going on.

This I think is a matter of perspective. Some would say "OK, I see what the defaults are, I can easily scan the `TEST_SUITE` entries to see which ones deviate from the defaults. Clear."

> The `dataset`, `config`, and `steps` keys should not be optional.

If they have a default value, they're not really optional though? Maybe this is just semantics (optional for user to type out, vs optional to exist in the dict at all), but I see value in *not* forcing people to type out `ds003392` multiple times (in key, dataset, and config) if it's possible to derive it from the key.

> without a `TypedDict`, my IDE doesn't provide me any help

Fair point.

Reply:
> as they reduce flexibility should we need it

I don't see how this is the case by providing sensible defaults?

> I specifically designed this here API to be very explicit and, yes, maybe a bit verbose, but with the advantage that no magic happens e.g. in terms of how the configuration files are located in the filesystem etc. ... The dataset, config, and steps keys should not be optional

To generalize your idea here, I think you could (nearly) equivalently argue in the spirit of explicitness/verboseness that functions should also have no default arguments, and users should always have to pass them. I don't think it's really sustainable or best practice. If something is the case often enough (maybe 50%+ of the time?), you should give it a it default and use it.

Coming back to this case, take `'config'` -- it is the same for every dataset except some ERP_CORE variants. It seems like a waste to specify it every time except for those cases. In this PR, it's immediately clear (and more clear on than on `main`) that the same config is reused in each of these cases, just with a new `task`.

And for the changes here, I'm able to just add this line:
```
    'ds004107`: {},
```
and everything "just works", and it's clear why with the comment above the dict (just like a function signature). In `main` I have to add a bunch of redundant stuff.

In the end, though, it's about balancing DRY vs explicitness vs risk of error (in different ways!), so I think we'll have to agree to disagree on how we should strike the balance here.

Maybe @agramfort can vote on whether 1) these three keys need to be obligatory, or 2) if they should all be optional to the extent possible, and then we can move on?

> Keep it a TypedDict but make all keys optional. So you can have your cake, and I can eat mine too:

I've pushed a commit for this part at least so far.

Reply:
Ok, let's move on here … `total=False` makes me sufficiently happy because I still get help from my IDE when I wish to add a new dataset; and you can omit everything you don't need, unless I'm overlooking something here. Thanks everybody for sharing your thoughts!